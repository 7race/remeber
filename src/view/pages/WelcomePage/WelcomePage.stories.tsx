import { ComponentStory, ComponentMeta } from '@storybook/react';
import { WelcomePage } from './WelcomePage';

export default {
  title: 'WelcomePage',
  component: WelcomePage,
} as ComponentMeta<typeof WelcomePage>;

const Template: ComponentStory<typeof WelcomePage> = () => <WelcomePage />;

export const Default = Template.bind({});
