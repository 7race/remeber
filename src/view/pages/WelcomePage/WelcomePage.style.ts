import styled from 'styled-components';

export const WelcomePage = styled.div`
  display: grid;
  justify-items: center;
  grid-gap: 100px;
  margin-top: 30px;
  & img {
    width: 200px;
    margin-bottom: 100px;
  }

  @media ${({ theme }) => theme.media.tablet} {
    & img {
      width: 500px;
      margin-bottom: 0px;
    }
  }
`;
