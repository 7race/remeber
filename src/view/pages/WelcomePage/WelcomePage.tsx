import Confetti from 'react-confetti';
import { P } from '@atoms/P';
import { Button } from '@atoms/Button';
import { meow } from '@static';
import { useWindowSize } from '@hooks/useWindowSize';
import * as S from './WelcomePage.style';

export const WelcomePage = () => {
  const { width, height } = useWindowSize();

  return (
    <>
      <Confetti width={width} height={height} />
      <S.WelcomePage>
        <P variant='h1'>Welcome</P>
        <img src={meow} alt='cat' />
        <Button variant='cancel' outlined>
          Log out
        </Button>
      </S.WelcomePage>
    </>
  );
};
