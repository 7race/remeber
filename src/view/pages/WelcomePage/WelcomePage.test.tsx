import { render } from '@tests-utils';
import { WelcomePage } from './WelcomePage';

describe('welcome page', () => {
  it('correct display', () => {
    const { container } = render(<WelcomePage />);
    expect(container).toMatchSnapshot();
  });
});
