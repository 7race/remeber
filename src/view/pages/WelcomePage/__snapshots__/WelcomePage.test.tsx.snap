// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome page correct display 1`] = `
<div>
  <canvas
    height="768"
    style="z-index: 2; position: absolute; pointer-events: none; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    width="1024"
  />
  <div
    class="sc-dkzDqf kPZTnG"
  >
    <p
      class="sc-bczRLJ Gujvk"
    >
      Welcome
    </p>
    <img
      alt="cat"
      src="cat.png"
    />
    <div
      class="react-ripples"
      style="position: relative; display: inline-flex; overflow: hidden;"
    >
      <button
        class="sc-gsnTZi fJxKCx"
      >
        Log out
      </button>
      <s
        style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
      />
    </div>
  </div>
</div>
`;
